client/packages/core/src/instaql.js

```diff
@@ -258,21 +260,33 @@ function extendObjects(makeVar, store, { etype, level, form }, objects) {
   return Object.entries(objects).map(([eid, parent]) => {
     const childResults = children.map((label) => {
       try {
-        const [nextEtype, nextLevel, join] = makeJoin(
+        const [nextEtype, nextLevel, join, attr, dir] = makeJoin(
           makeVar,
           store,
           etype,
           level,
           label,
           eid,
         );
-        const child = queryOne(store, {
+        const childrenArray = queryOne(store, {
           etype: nextEtype,
           level: nextLevel,
           form: form[label],
           join,
         });
-        return { [label]: child };
+
+        let children = childrenArray;
+        if (store.schema) {
+          const isForwardCardinalityOne =
+            dir === "forward" && attr.cardinality === "one";
+          const isReverseCardinalityOne = dir === "reverse" && attr["unique?"];
+
+          if (isForwardCardinalityOne || isReverseCardinalityOne) {
```

Comment:
nit: I would call this `isReverseUnique`. I do think `cardinality` is a good name for what we want, but would be nice to keep it with it's current meaning for now.